documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler-implement.adoc

```diff
@@ -156,6 +156,60 @@ It is therefore mandatory to use `CompletableFuture` APIs that are invoked even
 Failing to do so may result in the `Handler` `callback` parameter to never be completed, causing the request processing to hang forever.
 ====
 
+[[pg-server-http-handler-impl-request-listeners]]
+====== `Request` Listeners
+
+Application may add listeners to the `Request` object to be notified of particular events happening during the request/response processing.
+
+`Request.addIdleTimeoutListener(Predicate<TimeoutException>)` allows you to add an idle timeout listener, which is invoked when an idle timeout period elapses during the request/response processing, if the idle timeout event is not notified otherwise.
```

Comment:
Is there an e.g. of how an idle timeout event can be otherwise notified?

Is that really the case?   Isn't the idle listener always called to ask if the idle timeout can be ignored or not?

Reply:
The examples of `otherwise notified` are detail in the next paragraph as a bullet list.

It is really the case that the listener is not invoked if otherwise notified.
A read will receive a transient failure and decide to ignore it.
A write is likely fatal, but only fails the write callback (although the write cannot be resumed, and further writes will likely fail).
So we need to invoke the listener only if there was nothing pending.